web/src/components/DateRangeSelector.tsx

```diff
@@ -0,0 +1,68 @@
+import { Dialog, IconButton, InputAdornment, TextField } from '@mui/material'
+import * as materialIcons from '@mui/icons-material'
+import DateRangePickerWrapper from 'components/dateRangePicker/DateRangePickerWrapper'
+import { DateRange } from 'components/dateRangePicker/types'
+import { isUndefined } from 'lodash'
+import { DateTime } from 'luxon'
+import React, { useState } from 'react'
+
+export interface DateRangeSelectorProps {
+  dateRange?: DateRange
+  minDate?: Date
+  maxDate?: Date
+  maxDayOffset?: number
+  dateDisplayFormat: string
+  size: 'small' | 'medium'
+  label: string
+  setDateRange: (newDateRange: DateRange) => void
+}
+
+const DateRangeSelector = ({ dateRange, dateDisplayFormat, size, label, setDateRange }: DateRangeSelectorProps) => {
+  const startDate = dateRange && dateRange.startDate ? dateRange.startDate : undefined
+  const endDate = dateRange && dateRange.endDate ? dateRange.endDate : undefined
+
+  const [dateRangePickerOpen, setDateRangePickerOpen] = useState<boolean>(false)
+  const toggleDateRangePicker = () => setDateRangePickerOpen(!dateRangePickerOpen)
+
+  return (
+    <React.Fragment>
+      <TextField
+        data-testid="date-range-picker-text-field"
+        size={size}
+        variant="outlined"
+        disabled={true}
+        label={label}
+        onClick={() => setDateRangePickerOpen(!dateRangePickerOpen)}
+        value={
+          isUndefined(dateRange) || isUndefined(dateRange.startDate) || isUndefined(dateRange.endDate)
+            ? ''
+            : `${DateTime.fromJSDate(dateRange.startDate).toFormat(dateDisplayFormat).trim()} - ${DateTime.fromJSDate(
```

Comment:
Curious about .trim(). Does `toFormat()` add whitespace or line terminations? Or is the concern that the `dateDisplayFormat` might introduce these?

Reply:
No idea, I can remove it

Reply:
Removed `trim`, I didn't see any difference in how the string renders, it was probably a historical defensive measure.